docs/core/logging.md

```diff
@@ -377,320 +608,532 @@ for known event sources, where either a request ID or X-Ray Trace ID are present
-            ...
+            // ...
+            LoggingUtils.appendKey("cardNumber", card.getId());
+            LOGGER.info("Updating card information");
+            // ...
         }
     }
     ```
+
 === "#1 Request"
 
-    ```json hl_lines="11"
-	{
-		"level": "INFO",
-	  	"message": "Collecting payment",
-		"timestamp": "2021-05-03 11:47:12,494+0200",
-	  	"service": "payment",
-	  	"coldStart": true,
-	  	"functionName": "test",
-	  	"functionMemorySize": 128,
-	  	"functionArn": "arn:aws:lambda:eu-west-1:12345678910:function:test",
-	  	"function_request_id": "52fdfc07-2182-154f-163f-5f0f9a621d72",
-        "specialKey": "value"
-	}
+    ```json hl_lines="7"
+    {
+      "level": "INFO",
+      "message": "Updating card information",
+      "service": "card",
+      "timestamp": "2023-12-01T14:49:19.293Z",
+      "xray_trace_id": "1-6569f266-4b0c7f97280dcd8428d3c9b5",
+      "cardNumber": "6818 8419 9395 5322"
+    }
     ```
 
 === "#2 Request"
 
-    ```json
-	{
-		"level": "INFO",
-	  	"message": "Collecting payment",
-		"timestamp": "2021-05-03 11:47:12,494+0200",
-	  	"service": "payment",
-	  	"coldStart": true,
-	  	"functionName": "test",
-	  	"functionMemorySize": 128,
-	  	"functionArn": "arn:aws:lambda:eu-west-1:12345678910:function:test",
-	  	"function_request_id": "52fdfc07-2182-154f-163f-5f0f9a621d72"
-	}
+    ```json hl_lines="7"
+    {
+      "level": "INFO",
+      "message": "Updating card information",
+      "service": "card",
+      "timestamp": "2023-12-01T14:49:20.213Z",
+      "xray_trace_id": "2-7a518f43-5e9d2b1f6cfd5e8b3a4e1f9c",
+      "cardNumber": "7201 6897 6685 3285"
+    }
     ```
 
-## Override default object mapper
-
-You can optionally choose to override default object mapper which is used to serialize lambda function events. You might
-want to supply custom object mapper in order to control how serialisation is done, for example, when you want to log only
-specific fields from received event due to security.
+???+ tip "Additional keys are based on the MDC"
+    `clearState` is based on `MDC.clear()`. State clearing is automatically done at the end of the execution of the handler if set to `true`.
 
-=== "App.java"
 
-    ```java hl_lines="9 10"
-    /**
-     * Handler for requests to Lambda function.
-     */
-    public class App implements RequestHandler<APIGatewayProxyRequestEvent, APIGatewayProxyResponseEvent> {
-    
-        Logger log = LogManager.getLogger(App.class);
-
-        static {
-            ObjectMapper objectMapper = new ObjectMapper();
-            LoggingUtils.defaultObjectMapper(objectMapper);
-        }
-    
-        @Logging(logEvent = true)
-        public APIGatewayProxyResponseEvent handleRequest(final APIGatewayProxyRequestEvent input, final Context context) {
-            ...
-        }
-    }
-    ```
+# Advanced
 
 ## Sampling debug logs
 
 You can dynamically set a percentage of your logs to **DEBUG** level via env var `POWERTOOLS_LOGGER_SAMPLE_RATE` or
```

Comment:
"You can dynamically set a percentage of your logs at **DEBUG** level to be included in the logger output ..." or so

Reply:
```suggestion
You can dynamically set a percentage of your logs to`DEBUG` level to be included in the logger output, regardless of configured log leve, using the`POWERTOOLS_LOGGER_SAMPLE_RATE` environment variable or
```